janitor/functions/case_when.py

```diff
@@ -1,14 +1,17 @@
 from itertools import count
 from pandas.core.common import apply_if_callable
-from pandas.api.types import is_list_like
+from typing import Any, Optional
 import pandas_flavor as pf
 import pandas as pd
+from pandas.core.dtypes.inference import is_array_like
 
 from janitor.utils import check
 
 
 @pf.register_dataframe_method
-def case_when(df: pd.DataFrame, *args, column_name: str) -> pd.DataFrame:
+def case_when(
+    df: pd.DataFrame, *args, default: Optional[Any] = 0, column_name: str
```

Comment:
```suggestion
    df: pd.DataFrame, *args, *, default: Optional[Any] = 0, column_name: str
```

I suggest using positional argument syntax.
It may avoid some hidden problems since `case_when` could receive `*args`.

Reply:
since args is already variable arg, this is fine, and no issues should arise